File: .agent/skills/cnma-cap-backend/templates/node/srv/src/domain/__feature__/handler/BaseHandler.js
import { ApplicationService } from '@sap/cds/apis/services';

/**
 * BaseHandler - Abstract handler with common utilities.
 * SRP: Handler only registers events, logic goes to events/.
 */
export abstract class BaseHandler {
    constructor(protected readonly srv: ApplicationService) {}

    protected getUserId(req: any): string {
        return req.user?.id || 'anonymous';
    }

    protected getTenantId(req: any): string {
        return req.tenant || 'default';
    }
}

File: .agent/skills/cnma-cap-backend/templates/java/srv/src/main/java/cnma/{{module_name}}/domain/__feature__/handler/BaseHandler.java
package cnma.{{module_name}}.domain.__feature__.handler;

import org.springframework.stereotype.Component;

/**
 * BaseHandler - Abstract handler with common utilities.
 * Follows SRP: Handlers only handle event routing, delegate to service.
 */
@Component
public abstract class BaseHandler {

    protected String getUserId(Object user) {
        if (user instanceof cnma.{{module_name}}.infrastructure.security.UserContext) {
            return ((cnma.{{module_name}}.infrastructure.security.UserContext) user).getId();
        }
        return "anonymous";
    }

    protected String getTenantId(Object user) {
        if (user instanceof cnma.{{module_name}}.infrastructure.security.UserContext) {
            return ((cnma.{{module_name}}.infrastructure.security.UserContext) user).getTenant();
        }
        return "default";
    }
}